server/src/main/java/org/opensearch/action/admin/cluster/shards/TransportCatShardsAction.java

```diff
@@ -118,6 +122,11 @@ public void onResponse(ClusterStateResponse clusterStateResponse) {
                         indicesStatsRequest.setShouldCancelOnTimeout(true);
                         indicesStatsRequest.all();
                         indicesStatsRequest.indices(indices);
+                        // Since the indices for paginated query are already concrete and have been filtered to
+                        // only consider OPEN indices, invoking IndexNameExpressionResolver should be avoided.
+                        if (paginationStrategy != null) {
+                            indicesStatsRequest.skipIndexNameResolver(true);
```

Comment:
this flag looks hacky

Reply:
@shwetathareja Its more of an optimisation than a hack. Even if we run the resolver, it would result in same set of indices, this is just being done to avoid multiple passes over the indices list and thereby improve on latencies for each paginated query.